File: inventory.py
from flask import Blueprint, request, jsonify
from database import db_connection

inventory_bp = Blueprint('inventory', __name__)

db = db_connection()
cursor = db.cursor(dictionary=True)

# Get All Inventory Items
@inventory_bp.route('/', methods=['GET'])
def get_inventory():
    cursor.execute("SELECT * FROM inventory")
    inventory = cursor.fetchall()
    return jsonify(inventory), 200

# Add New Inventory Item
@inventory_bp.route('/', methods=['POST'])
def add_inventory_item():
    data = request.json
    cursor.execute("INSERT INTO inventory (product_name, quantity, reorder_level, supplier_id) VALUES (%s, %s, %s, %s)",
                   (data['product_name'], data['quantity'], data['reorder_level'], data['supplier_id']))
    db.commit()
    return jsonify({"message": "Inventory item added successfully"}), 201

# Restock Inventory Item
@inventory_bp.route('/<int:item_id>/restock', methods=['PUT'])
def restock_inventory_item(item_id):
    cursor.execute("UPDATE inventory SET quantity = quantity + 10 WHERE id = %s", (item_id,))
    db.commit()
    return jsonify({"message": "Inventory item restocked successfully"}), 200

File: inventory.js
document.addEventListener("DOMContentLoaded", function () {
    fetchInventory();
});

function fetchInventory() {
    fetch("http://127.0.0.1:5001/inventory")
        .then(response => response.json())
        .then(data => {
            let inventoryTable = document.getElementById("inventoryTable");
            inventoryTable.innerHTML = "";

            if (data.length === 0) {
                inventoryTable.innerHTML = "<tr><td colspan='6'>No inventory records found.</td></tr>";
                return;
            }

            data.forEach(item => {
                let row = `
                    <tr>
                        <td>${item.id}</td>
                        <td>${item.product_name}</td>
                        <td>${item.quantity}</td>
                        <td>${item.reorder_level}</td>
                        <td>${item.supplier_id}</td>
                        <td>
                            <button class="restock-btn" onclick="restockItem(${item.id})">Restock</button>
                        </td>
                    </tr>
                `;
                inventoryTable.innerHTML += row;
            });
        })
        .catch(error => console.error("Error fetching inventory:", error));
}

function restockItem(itemId) {
    fetch(`http://127.0.0.1:5001/inventory/${itemId}/restock`, { method: "PUT" })
        .then(response => response.json())
        .then(data => {
            alert(data.message);
            fetchInventory();
        })
        .catch(error => console.error("Error restocking item:", error));
}
